### PR TITLE
remove dead if statement in Module['read']

### PR DESCRIPTION
**Pull request by ralphtheninja**
Return buffer if binary, otherwise return `.toString()`

Also see discussion in https://github.com/kripken/emscripten/pull/4168

### CODE REVIEW

File: src/shell.js [hunk 1]
```diff
@@ -83,16 +83,9 @@ if (ENVIRONMENT_IS_NODE) {
...
     filename = nodePath['normalize'](filename);
     var ret = nodeFS['readFileSync'](filename);
-    // The path is absolute if the normalized version is the same as the resolved.
-    if (!ret && filename != nodePath['resolve'](filename)) {
-      filename = path.join(__dirname, '..', 'src', filename);
-      ret = nodeFS['readFileSync'](filename);
-    }
-    if (ret && !binary) ret = ret.toString();
-    return ret;
+    return (binary ? ret : ret.toString());
```
Comment (kripken) [inline src/shell.js]: No need for the outer parentheses.

**Comment by kripken** [comment]
Please also add yourself to AUTHORS.

**Comment by ralphtheninja** [comment]
> Please also add yourself to AUTHORS.

Amended commit and rebased onto latest incoming

**Comment by kripken** [comment]
Thanks!

**Pull request merged**